### ISSUE TITLE
Unable to build LLVM on macOS

### ISSUE THREAD

**Issue by adamjstewart** [spack/spack#4985]
I'm seeing the following error message when trying to build LLVM on macOS:
```
error: The specified item could not be found in the keychain.
```
Full build log [here](https://github.com/LLNL/spack/files/1203281/spack-build.txt). Any ideas?

**Comment by krcb** [comment]
This appears to be an issue related to code signing, which is required for building lldb (part of llvm) on the MacOS. It can be resolved by following the build instructions given on the lldb website for MacOS:

https://lldb.llvm.org/build.html#BuildingLldbOnMacOSX

which points to the user to lldb/docs/code-signing.txt in the lldb checkout, which under spack is located in the tools directory. As I'm not a spack developer, I'm not sure if this is documented for this package somewhere already and if not where this should be documented.

**Comment by arh-oxbotica** [comment]
FWIW the homebrew LLVM package performs an explicit check for the ability to sign code. It will refuse to build llvm with the `lldb` option turned on if you don't have the ability to do code signing.

https://github.com/Homebrew/homebrew-core/blob/8f740c21cdbcc94d605226e5effe1a09fe667856/Formula/llvm.rb

I'm new to Spack, so don't know how easy this would be to implement. Just wanted to flag up the fact that another community has already thought about and solved this problem.

**Comment by adamjstewart** [comment]
This would be pretty easy to implement in Spack. I can try adding it if I find any free time.

**Comment by alalazo** [comment]
Should be addressed by #7012. Please reopen if this is still an issue.
